### PR TITLE
feat: add provenance-name to the outputs

### LINKED ISSUE #1842: [feature] SLSA 3 Go Builder add provenance name to the workflow outputs
**Is your feature request related to a problem? Please describe.**
This is a feature.

**Describe the solution you'd like**

At the time of writing this, SLSA 3 Go Builder does not have the provenance name generated during the workflow within its outputs[^1]. So, it'd be better to provide an output that specifies the provenance name.

**Additional context**
I'm willing to do this, please assign it to me. cc @laurentsimon 

[^1]: https://github.com/slsa-framework/slsa-github-generator/blob/667cb7e0d6b66bd22435d20e23d9ecaf82bd7682/.github/workflows/builder_go_slsa3.yml#L82

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes #1842

### CODE REVIEW

File: .github/workflows/builder_go_slsa3.yml [hunk 1]
```diff
@@ -83,6 +83,9 @@ on:
       go-binary-name:
         description: "The name of the generated binary uploaded to the artifact registry."
         value: ${{ jobs.build-dry.outputs.go-binary-name }}
+      provenance-name:
+        description: "The artifact name of the signed provenance. (A file with the intoto.jsonl extension)."
+        value: ${{ jobs.provenance.outputs.go-provenance-name}}
```
Comment (ianlewis) [inline .github/workflows/builder_go_slsa3.yml:88]: nit: add extra space
```suggestion
        value: ${{ jobs.provenance.outputs.go-provenance-name }}
```

**Review by ianlewis — COMMENTED** [review]
Thanks!

- [x] Can you add an note to the CHANGELOG.md for the new output?
- [x] Can you add documention for the output to the README for the Go workflow?

File: .github/workflows/builder_go_slsa3.yml (hunk 1 above)
Comment (developer-guy) [reply .github/workflows/builder_go_slsa3.yml:88]: fixed, thank you.

Comment (laurentsimon) [reply .github/workflows/builder_go_slsa3.yml:88]: I dont see the update

**Comment by laurentsimon** [comment]
You'll need to update code to populate the provenance filename, probably in https://github.com/slsa-framework/slsa-github-generator/blob/f46e50b9e21bca4aacbd305f62e1ff63c55133f0/internal/builders/go/pkg/build.go#L167-L183

File: .github/workflows/builder_go_slsa3.yml (hunk 1 above)
Comment (developer-guy) [reply .github/workflows/builder_go_slsa3.yml]: I forgot to push 🤦

**Comment by developer-guy** [comment]
> You'll need to update code to populate the provenance filename, probably in
> 
> https://github.com/slsa-framework/slsa-github-generator/blob/f46e50b9e21bca4aacbd305f62e1ff63c55133f0/internal/builders/go/pkg/build.go#L167-L183

is this necessary ? 🤔

**Comment by developer-guy** [comment]
because I updated the tests according to the changes to see whether it is working, and it worked.

**Comment by laurentsimon** [comment]
> > You'll need to update code to populate the provenance filename, probably in
> > https://github.com/slsa-framework/slsa-github-generator/blob/f46e50b9e21bca4aacbd305f62e1ff63c55133f0/internal/builders/go/pkg/build.go#L167-L183
> 
> is this necessary ? 🤔

nope, I delete my comment. I mis-read ^^

**Comment by laurentsimon** [comment]
You need to run the toc CLI as in the failing pre-submit

**Comment by laurentsimon** [comment]
```
$ npm install markdown-toc
$ ./node_modules/markdown-toc/cli.js --bullets="-" -i README.md
```

should work. Let me know.

File: .github/workflows/builder_go_slsa3.yml [hunk 2]
```diff
@@ -83,6 +83,9 @@ on:
       go-binary-name:
         description: "The name of the generated binary uploaded to the artifact registry."
         value: ${{ jobs.build-dry.outputs.go-binary-name }}
+      provenance-name:
```
Comment (laurentsimon) [inline .github/workflows/builder_go_slsa3.yml:86]: should this be `go-provenance-name` for consistency with `go-binary-name`?

**Review by ianlewis — APPROVED** [review]
LGTM.

I went ahead and made the changes based on @laurentsimon's comments since he went ahead and committed changes to our e2e tests.

**Pull request merged**